collectors/googlestackdriver/collector.js

```diff
@@ -92,6 +104,20 @@ timestamp < "${state.until}"`;
             })
             .catch(err => {
                 console.error(`GSTA000003 err in collection ${err}`);
+                
+                // Stackdriver Logging api has some rate limits that we might run into.
+                // If we run inot a rate limit error, instead of returning the error,
+                // we return the state back to the queue with an additional second added, up to 10
+                // https://cloud.google.com/logging/quotas
+                if(err.match(/RESOURCE_EXHAUSTED/)){
```

Comment:
maybe we get some error code here we can use instead heavy `match()`?

Reply:
Good point. Yeah it looks like there are actually error codes in these errors. i think they show up nicely as strings becuase google was fancy with the way they are stringified. 

This mathc doesnt seem to work anyway. looks like these are more complex than just stirngs.